platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/tipset/TipsetEventCreatorImpl.java

```diff
@@ -268,10 +273,11 @@ private GossipEvent createEventByOptimizingAdvancementWeight() {
             }
         }
 
-        if (lastSelfEvent != null && bestOtherParent == null) {
-            // There exist no parents that can advance consensus, and this is not our first event.
-            // The only time it's ok to create an event with no other parent is when we are creating
-            // our first event.
+        if (bestOtherParent == null
+                && (minimumGenerationNonAncient > GENERATION_UNDEFINED + 1 || lastSelfEvent != null)) {
```

Comment:
I think the logic should be: "don't create ancient events"
So even if we have parents, but they are ancient, we should not create an event

Reply:
`childlessOtherEventTracker.getChildlessEvents()` will never return ancient events, so we are guaranteed that any parent we choose will have a generation that produces a non-ancient child.